app_dart/lib/src/request_handlers/github/webhook_subscription.dart

```diff
@@ -253,17 +253,19 @@ class GithubWebhookSubscription extends SubscriptionHandler {
   Future<void> _checkForTests(PullRequestEvent pullRequestEvent) async {
     final PullRequest pr = pullRequestEvent.pullRequest!;
     final String? eventAction = pullRequestEvent.action;
-    final String repo = pr.base!.repo!.fullName.toLowerCase();
-    if (kNeedsTests.contains(repo)) {
-      final GitHub gitHubClient = await config.createGitHubClient(pullRequest: pr);
+    final RepositorySlug slug = pr.base!.repo!.slug();
+    final bool isTipOfTree = pr.base!.ref == Config.defaultBranch(slug);
+    final GitHub gitHubClient = await config.createGitHubClient(pullRequest: pr);
+    await _validateRefs(gitHubClient, pr);
+    if (kNeedsTests.contains(slug) && isTipOfTree) {
       try {
-        await _validateRefs(gitHubClient, pr);
-        if (repo == 'flutter/flutter') {
-          await _applyFrameworkRepoLabels(gitHubClient, eventAction, pr);
-        } else if (repo == 'flutter/engine') {
-          await _applyEngineRepoLabels(gitHubClient, eventAction, pr);
-        } else if (repo == 'flutter/plugins' || repo == 'flutter/packages') {
-          await _applyPackageTestChecks(gitHubClient, eventAction, pr);
+        switch (slug.name) {
+          case 'flutter':
+            return _applyFrameworkRepoLabels(gitHubClient, eventAction, pr);
+          case 'engine':
+            return _applyEngineRepoLabels(gitHubClient, eventAction, pr);
+          case 'packages':
```

Comment:
Umm why did we remove the plugins repo

Reply:
See https://flutter.dev/go/flutter-plugins-repo-migration

The plugins repo was migrated to packages. There shouldn't be any migrations for the plugins other than removing outdated tests.

Reply:
thanks for explaining Casey!

sorry for bothering Casey when Casey is on vacation